io.openems.edge.bridge.modbus/src/io/openems/edge/bridge/modbus/sunspec/AbstractOpenemsSunSpecComponent.java

```diff
@@ -37,6 +40,11 @@
  */
 public abstract class AbstractOpenemsSunSpecComponent extends AbstractOpenemsModbusComponent {
 
+	/**
+	 * Limit of a task length in j2mod.
+	 */
+	private static final int MAXIMUM_TASK_LENGTH = 126;
```

Comment:
If this is really a hard limit, we should actually put this constant deeper into our Modbus Bridge. I am not able to find 126 as an explicit limit in j2mod (apart from https://github.com/steveohara/j2mod/blob/development/src/main/java/com/ghgande/j2mod/modbus/msg/WriteFileRecordRequest.java#L201, which appears to be exactly this hard-coded limit), but there is `256` as a maximum message length in https://github.com/steveohara/j2mod/blob/development/src/main/java/com/ghgande/j2mod/modbus/Modbus.java#L256C19-L256C19

I am not saying your approach is wrong, just:
- it would be perfect if we could use a constant from j2mod directly
- we should validate all Tasks e.g. in `defineModbusProtocol()` to throw an exception as early as possible

Reply:
I was also thinking about this, as other modbus components might run into the same problem, and it's really hard to find the error if you don't know the limit (I was lucky I did, and I am grateful to whoever wrote it in the comment in AbstactEdge2Edge).